packages/mds-cache/index.ts

```diff
@@ -105,6 +126,70 @@ async function info() {
   return data
 }
 
+async function hget(key: string, field: UUID): Promise<CachedItem | CachedHashItem | null> {
+  const flat = await (await getClient()).hgetAsync(key, field)
+  if (flat) {
+    return unflatten(flat)
+  }
+  return null
+  // throw new Error(`${field} not found in ${key}`)
+}
+
+async function hgetall(key: string): Promise<CachedItem | CachedHashItem | null> {
+  const flat = await (await getClient()).hgetallAsync(key)
+  if (flat) {
+    return unflatten(flat)
+  }
+  return null
+  // throw new Error(`${key} not found`)
+}
+
+async function readDeviceState(field: UUID): Promise<StateEntry | null> {
+  const deviceState = await hget('device:state', field)
```

Comment:
Might be good to have an `Enum` that contains all possible cache keys so we catch typos.

Reply:
String literal instead of an enum, but yes.